src/main/java/com/stefancooper/SpigotUHC/events/BaseEvents.java

```diff
@@ -206,4 +226,196 @@ public void onChatMessage(AsyncPlayerChatEvent event) {
+
+        // Helmet slot index in player inventory is 5
+        if (event.getInventorySlots().contains(39)) {
+            config.getManagedResources().runTaskLater(() -> {
+                final ItemStack helmet = player.getInventory().getHelmet();
+
+                if (AdditionalEnchants.isNightVisionGoggles(helmet)) {
+                    if (!player.hasPotionEffect(PotionEffectType.NIGHT_VISION)) {
+                        player.addPotionEffect(new PotionEffect(
+                                PotionEffectType.NIGHT_VISION,
+                                Integer.MAX_VALUE,
+                                0,
+                                true,
+                                false
+                        ));
+                    }
+                } else {
+                    player.removePotionEffect(PotionEffectType.NIGHT_VISION);
+                }
+            }, 1L);
+        }
+    }
+
+    @EventHandler
+    public void onPlayerJoin(PlayerJoinEvent event) {
+        applyNightVisionIfWearingHelmet(event.getPlayer());
+    }
+
+    private void applyNightVisionIfWearingHelmet(Player player) {
+        final ItemStack helmet = player.getInventory().getHelmet();
+        if (AdditionalEnchants.isNightVisionGoggles(helmet)) {
+            player.addPotionEffect(new PotionEffect(
+                    PotionEffectType.NIGHT_VISION,
+                    Integer.MAX_VALUE,
+                    0,
+                    true,
+                    false
+            ));
+        }
+    }
+
+    @EventHandler
+    public void onShieldBlock(EntityDamageByEntityEvent event) {
+        if (!(event.getEntity() instanceof Player defender)) return;
+        if (!(event.getDamager() instanceof LivingEntity attacker)) return;
+        if (!defender.isBlocking()) return;
+
+        // Check both hands for a shield
+        ItemStack shield = null;
+        ItemStack offHand = defender.getInventory().getItemInOffHand();
+        ItemStack mainHand = defender.getInventory().getItemInMainHand();
+
+        if (offHand != null && offHand.getType() == Material.SHIELD) {
+            shield = offHand;
+        } else if (mainHand != null && mainHand.getType() == Material.SHIELD) {
+            shield = mainHand;
+        }
+
+        if (shield == null || !shield.hasItemMeta()) return;
+
+        ItemMeta meta = shield.getItemMeta();
+        if (meta == null || !meta.hasLore()) return;
+
+        int knockbackLevel = getEnchantLevelFromLore(meta, "Knockback");
+        int thornsLevel = getEnchantLevelFromLore(meta, "Thorns");
+
+        // Apply Thorns damage
+        if (thornsLevel > 0) {
+            double damage = 1.0 + 0.5 * (thornsLevel - 1);
+            attacker.damage(damage, defender);
+            attacker.getWorld().spawnParticle(
+                    Particle.DAMAGE_INDICATOR,
+                    attacker.getLocation().add(0, 1, 0),
+                    5 + 2 * thornsLevel

```

Comment:
some of these calculations are quite specific - i am okay with that but i think it needs documenting for future (and for my own curiosity :) )